lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,353 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module to contain Psychrometric Calculations."""
+
+import warnings
+from cf_units import Unit
+import improver.constants as cc
+import numpy as np
+
+from improver.psychrometric_calculations import svp_table
+
+
+class Utilities(object):
+
+    """
+    Utilities for pschrometric calculations.
+    """
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<Utilities>')
+        return result
+
+    @staticmethod
+    def specific_heat_of_moist_air(mixing_ratio):
+        """
+        Calculate the specific heat capacity for moist air by combining that of
+        dry air and water vapour in proportion given by the specific humidity.
+
+        Args:
+            mixing_ratio : iris.cube.Cube
+                Cube of specific humidity (fractional).
+        Returns:
+            iris.cube.Cube
+                Specific heat capacity of moist air (J kg-1 K-1).
+        """
+        specific_heat = ((-1.*mixing_ratio + 1.) * cc.U_CP_DRY_AIR +
+                         mixing_ratio * cc.U_CP_WATER_VAPOUR)
+        specific_heat.rename('specific_heat')
+        return specific_heat
+
+    @staticmethod
+    def latent_heat_of_condensation(temperature_input):
+        """
+        Calculate a temperature adjusted latent heat of condensation for water
+        vapour using the relationship employed by the UM.
+
+        Args:
+            temperature : iris.cube.Cube
+                A cube of air temperatures (Celsius, converted if not).
```

Comment:
This argument should be temperature_input not temperature

Reply:
:white_check_mark: